src/rgw/rgw_sync_module_es.cc

```diff
@@ -432,18 +523,33 @@ class RGWElasticInitConfigCBCR : public RGWCoroutine {
   int operate() override {
     reenter(this) {
       ldout(sync_env->cct, 0) << ": init elasticsearch config zone=" << sync_env->source_zone << dendl;
+      yield call(new RGWReadRESTResourceCR<ESInfo> (sync_env->cct,
+                                                                conf->conn.get(),
+                                                                sync_env->http_manager,
+                                                                "/", nullptr, &es_info));
+      if (retcode < 0) {
+        return set_cr_error(retcode);
+      }
+
       yield {
         string path = conf->get_index_path();
+        ldout(sync_env->cct, 5) << "got elastic version=" << es_info.version.to_str() << dendl;
 
         es_index_settings settings(conf->num_replicas, conf->num_shards);
         es_index_mappings mappings;
+        if (es_info.version >= ESVersion(5,0)) {
+          ldout(sync_env->cct, 0) << "elasticsearch: using text type for string index mappings " << dendl;
+          mappings.string_type = ESType::Text;
```

Comment:
maybe Keyword is enough, we don't need analyse most field

Reply:
I guess for id like fields we can go for keyword, text is still better for names, like the object/bucket names since keywords expect the exact search term